packages/theme/components/Checkout/PaymentMethod/Stripe.vue

```diff
@@ -16,64 +24,73 @@ export default {
   },
 
   setup(props, { emit }) {
+    const { $config } = useContext();
     const { $spree } = useVSFContext();
+    const { cart } = useCart();
     const stripe = ref(null);
-    const card = ref(null);
-    const cardRef = ref(null);
-    const areIntentsEnabled = computed(() => props.method.preferences?.intents);
+    const payment = ref(null);
+    const elements = ref(null);
+    const paymentRef = ref(null);
+    const errorRef = ref(null);
     const publishableKey = computed(() => props.method.preferences?.publishable_key);
 
     const savePayment = async () => {
       try {
-        const methodId = props.method.id;
-        const { token } = await stripe.value.createToken(card.value);
-
-        const payload = {
-          // eslint-disable-next-line camelcase
-          gateway_payment_profile_id: token.id,
-          number: token.card.last4,
-          month: token.card.exp_month,
-          year: token.card.exp_year,
-          name: token.card.name
-        };
-
-        await $spree.api.savePaymentMethod(methodId, payload);
-
-        if (areIntentsEnabled.value) {
-          const threeDSecureData = await $spree.api.getPaymentConfirmationData();
-          const confirmCardPaymentResponse = await stripe.value.confirmCardPayment(threeDSecureData.clientSecret, {});
-          const handlePaymentConfirmationResponse = await $spree.api.handlePaymentConfirmationResponse({
-            confirmationResponse: confirmCardPaymentResponse
-          });
-
-          if (!handlePaymentConfirmationResponse.success) {
-            throw new Error('Failed to confirm payment');
+        const orderId = orderGetters.getId(cart.value);
+        const { error } = await stripe.value.confirmPayment({
+          elements: elements.value,
+          // Note: If payment is successfully confirmed, user will be redirected to the return_url
```

Comment:
This is well explained in the documentation. I don't think we need this comment.

https://stripe.com/docs/js/payment_intents/confirm_payment?type=cardExpiry

Reply:
Not a problem. Agree that it is well explained in the documentation. Just wasn't sure how familiar others would be with the Stripe flow. Removed.